### PR TITLE
Clarify "uncaught runtime script error".

### PR DESCRIPTION
**Pull request by mattto**
The HTML standard returns Completion Records to convey the result
of running a script. Use that.

***
<a href="https://pr-preview.s3.amazonaws.com/mattto/ServiceWorker/pull/1346.html" title="Last updated on Oct 31, 2018, 7:06 AM GMT (88020ac)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/ServiceWorker/1346/7b85d13...mattto:88020ac.html" title="Last updated on Oct 31, 2018, 7:06 AM GMT (88020ac)">Diff</a>

### CODE REVIEW

**Comment by mattto** [comment]
HTML spec change: https://github.com/whatwg/html/pull/3907

**Comment by mattto** [comment]
The HTML spec will change use NormalCompletion(empty) when scripting is disabled in https://github.com/whatwg/html/pull/3907. Updated this PR to align with that.

File: docs/index.bs [hunk 1]
```diff
@@ -2481,8 +2482,8 @@ spec: webappsec-referrer-policy; urlPrefix: https://w3c.github.io/webappsec-refe
           1. Set |worker|'s [=script resource map=][|url|] to |response|.
       1. Set |worker|'s <a>script resource</a>'s <a>HTTPS state</a> to |httpsState|.
       1. Set |worker|'s <a>script resource</a>'s [=script resource/referrer policy=] to |referrerPolicy|.
-      1. Invoke <a>Run Service Worker</a> algorithm given |worker|, and with the *force bypass cache for importscripts flag* set if |job|'s [=job/force bypass cache flag=] is set.
-      1. If an uncaught runtime script error occurs during the above step, then:
+      1. Let |promise| be the result of invoking the [=Run Service Worker=] algorithm given |worker|, and with the *force bypass cache for importscripts flag* set if |job|'s [=job/force bypass cache flag=] is set.
+      1. [=upon fulfillment|Upon fulfillment=] of |promise| with value |evaluationStatus|, if |evaluationStatus|.\[[Value]] is empty or an [=abrupt completion=], then:
```
Comment (mattto) [inline docs/index.bs]: This should read "if evaluationStatus.[[Value]] is empty or evaluationStatus is an abrupt completion". I'll revise.

**Comment by mattto** [comment]
Revised. @jungkees  please take a look.

File: docs/index.bs [hunk 2]
```diff
@@ -2481,8 +2482,8 @@ spec: webappsec-referrer-policy; urlPrefix: https://w3c.github.io/webappsec-refe
           1. Set |worker|'s [=script resource map=][|url|] to |response|.
       1. Set |worker|'s <a>script resource</a>'s <a>HTTPS state</a> to |httpsState|.
       1. Set |worker|'s <a>script resource</a>'s [=script resource/referrer policy=] to |referrerPolicy|.
-      1. Invoke <a>Run Service Worker</a> algorithm given |worker|, and with the *force bypass cache for importscripts flag* set if |job|'s [=job/force bypass cache flag=] is set.
-      1. If an uncaught runtime script error occurs during the above step, then:
+      1. Let |promise| be the result of invoking the [=Run Service Worker=] algorithm given |worker|, and with the *force bypass cache for importscripts flag* set if |job|'s [=job/force bypass cache flag=] is set.
+      1. [=upon fulfillment|Upon fulfillment=] of |promise| with value |evaluationStatus|, if |evaluationStatus| is an [=abrupt completion=] or |evaluationStatus|.\[[Value]] is empty, then:
```
[the remark this answers is not in the document]
Comment (mattto) [reply docs/index.bs]: I see, I'd have to move everything into the substeps.

Yes exactly. The register/update steps start up the service worker and wait for a callback signalling whether execution was successful. If successful, they continue on. If unsuccessful, they abort.

**Comment by jungkees** [comment]
@mattto, sorry for the late response.

> I'm not sure how to fix the "not precise" problem

Rather than using a promise here, how about passing an optional callback steps to Run Service Worker? I think by that we won't have any semantical impact on call sites and also can avoid resolving a promise from different parallel thread.

**Comment by jungkees** [comment]
@mattto, similar to what we're doing with the "perform the fetch" steps to the importScripts(): https://w3c.github.io/ServiceWorker/#importscripts. Or something we had done with the [Handle Functional Event](https://www.w3.org/TR/2017/WD-service-workers-1-20171102/#handle-functional-event-algorithm) algorithm before having been replaced by Fire Functional Event.

**Comment by mattto** [comment]
@jungkees: I changed from a promise to a callback. Can you take another look please?

**Pull request merged**